deflect/ServerWorker.cpp

```diff
@@ -86,7 +86,10 @@ ServerWorker::~ServerWorker()
     // if other senders are still active / resp. the window gets closed if no
     // more senders contribute to it.
     if (!_streamId.isEmpty())
+    {
         emit removeStreamSource(_streamId, _sourceId);
+        emit removeObserver(_streamId, _sourceId);
```

Comment:
look like a bit brute force, maybe remember the stream type and emit just the correct signal?